bin/cron/ops_data_pull

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/properties'
+include ApplicationHelper
+include UsersHelper
+
+pd_scripts = Script.all.select { |script| !!script.pd }
+pd_scripts.each do |script|
+  # Get all users with any activity in the script
+  users = User.joins('left join user_scripts on user_scripts.user_id = users.id').where(user_scripts: {script_id: script})
+  headers = nil
+  data = users.map do |user|
+    row = {}
+    row.merge!({
+                   :'ID' => user.id,
+                   :'Ops First Name' => user.ops_first_name,
+                   :'Ops Last Name' => user.ops_last_name,
+                   :'Email' => user.email,
+                   :'District' => user.district.try(:name) || 'None'
+               })
+    user_progress = summarize_user_progress(script, user)
+    percent = percent_complete(script, user)
+    script.stages.each do |stage|
+      levels = Hash[stage.script_levels.map(&:level).map do |level|
```

Comment:
This section of code is a bit tricky. I think it's fine to take a couple more lines and spell things out a little bit more if it helps readability and efficiency.  e.g.

levels = {}
stage.script_levels.each.do |level|
  progress = user_progress[:levels][level.id]
  level[level_id.to_s] = (progress && (progress[:status] == 'perfect')) ? '1' : '0')
end